File: src/AceRental.Application/Reservations/Dtos/ReservationServicesDto.cs
using System.ComponentModel.DataAnnotations;
using AceRental.Application.Services.Dtos;

namespace AceRental.Application.Reservations.Dtos
{
    public class ReservationServicesDto
    {
        public Guid ReservationId { get; set; }
        public Guid ServiceId { get; set; }
        public int Quantity { get; set; }
        public decimal UnitPriceAtTimeOfBooking { get; set; } // Historisation du prix
        public ReservationDto Reservation { get; set; } = null!;
        public ServiceDto Service { get; set; } = null!;
    }
}

File: AceRental.FrontEnd/ClientApp/src/app/interfaces/reservation-services-dto.ts
import { ReservationDetailsDto } from "./reservation-details-dto";
import { ServiceDto } from "./service-dto";

export interface ReservationServicesDto {
    ReservationId?: string;
    ServiceId: string;
    Quantity: number;
    UnitPriceAtTimeOfBooking: number;
    Reservation?: ReservationDetailsDto;
    Service?: ServiceDto;
}
